### PR TITLE
2.0 restore subdomain for pagerduty ack retrievals - fixes #855

### LINKED ISSUE #855: Pagerduty gateway needs to know about subdomain
See [this comment](https://github.com/flapjack/flapjack/pull/821#issuecomment-91037179) on PR #821 which erroneously removes the subdomain from PagerDuty REST api URLs.

### PR DESCRIPTION
**Pull request by jessereynolds**
fixes #855 

subdomain is required by the pagerduty incidents api. it was removed in PR #821

### CODE REVIEW

File: lib/flapjack/gateways/pager_duty.rb [hunk 1]
```diff
@@ -184,26 +185,26 @@ def start
...
-                  Flapjack.redis.del(SEM_PAGERDUTY_ACKS_RUNNING)
-                end
+        # begin
+
+          Zermelo.redis = Flapjack.redis
+
+          loop do
+            @lock.synchronize do
+              # ensure we're the only instance of the PagerDuty acknowledgement check running (with a naive
+              # timeout of five minutes to guard against stale locks caused by crashing code) either in this
```
Comment (Hobbsee) [inline lib/flapjack/gateways/pager_duty.rb]: This text needs an update.

File: lib/flapjack/gateways/pager_duty.rb [hunk 2]
```diff
@@ -357,13 +356,13 @@ def pagerduty_acknowledgements(time, token, check_names)
           cumulative_incidents
         end
 
-        def pagerduty_acknowledgements_request(base_time, token, limit, offset)
+        def pagerduty_acknowledgements_request(base_time, subdomain, token, limit, offset)
           since_offset, until_offset = if @initial
             # the last week -> one hour in the future
-            [(60 * 60 * 24 * 7), (60 * 24)]
+            [(60 * 60 * 24 * 7), (60 * 60)]
```
Comment (Hobbsee) [inline lib/flapjack/gateways/pager_duty.rb]: Good catch on these.

**Comment by Hobbsee** [comment]
Looks good to me, after the comment update.

File: lib/flapjack/gateways/pager_duty.rb (hunk 1 above)
Comment (ali-graham) [inline lib/flapjack/gateways/pager_duty.rb]: As does the logic :( https://github.com/flapjack/zermelo/blob/master/lib/zermelo/locks/redis_lock.rb is used in `zermelo` for class-level locking, we could maybe use that class directly to set up something safer, with an appropriate timeout (it should release the lock even on exception, but nothing's going to protect against, e.g. `kill -9`)).

Comment (jessereynolds) [inline lib/flapjack/gateways/pager_duty.rb]: Comments need fixing, thanks @Hobbsee 

@ali-graham - what's wrong with the logic?

Comment (ali-graham) [inline lib/flapjack/gateways/pager_duty.rb]: It'd be nice to decouple Flapjack from talking to Redis directly, and this is one of the few places (in `master`, even more so in the `api_web` branch) that it still does. Logically it's fine as is (although as observed the expiry is a bit of a hack).

File: lib/flapjack/gateways/pager_duty.rb (hunk 2 above)
Comment (jessereynolds) [inline lib/flapjack/gateways/pager_duty.rb]: :-) 24 minutes per hour? :trollface:

File: lib/flapjack/gateways/pager_duty.rb (hunk 1 above)
Comment (jessereynolds) [inline lib/flapjack/gateways/pager_duty.rb]: @ali-graham ah right, yes. Agreed. Will create an issue, and entertain the idea of implementing it.

**Comment by ali-graham** [comment]
Looks good, merging. I'll remove the username/password from the specs as we'll use the 2.0 release to break compatibility with 1.0 and only support token.

**Pull request merged**